cloud/scope/cluster.go

```diff
@@ -91,9 +91,11 @@ func (s *ClusterScope) Close(ctx context.Context) error {
 	return s.PatchObject(ctx)
 }
 
-// AddFinalizer adds a finalizer and immediately patches the object to avoid any race conditions
+// AddFinalizer adds a finalizer if not present and immediately patches the
+// object to avoid any race conditions.
 func (s *ClusterScope) AddFinalizer(ctx context.Context) error {
-	controllerutil.AddFinalizer(s.LinodeCluster, infrav1alpha1.GroupVersion.String())
-
-	return s.Close(ctx)
+	if controllerutil.AddFinalizer(s.LinodeCluster, infrav1alpha1.GroupVersion.String()) {
```

Comment:
Can we add a comment here and on the other files explaining we only do Close if the finalizer was actually updated?